pkg/metadata/store.go

```diff
@@ -44,7 +44,7 @@ func UpdateMetadata(ctx context.Context, backend storage.Backend, meta *v1alpha1
 func resolveOperatorMetadata(ctx context.Context, operator v1alpha1.Operator, backend storage.Backend, insecure bool) (operatorMeta v1alpha1.OperatorMetadata, err error) {
 	operatorMeta.Catalog = operator.Catalog
 
-	resolver, err := containerdregistry.NewResolver(operator.PullSecret, insecure, nil)
+	resolver, err := containerdregistry.NewResolver("", insecure, nil)
```

Comment:
From all the changes in this PR, I gather that if users want to include any secrets (such as pull secret), they will need to include that as a part of Docker/Podman setting. The option of providing discrete secrets outside of the daemons is essentially gone. Would that be a problem with someone wants to provide pull secrets with the set of images but has no access to daemons (such as lacking of admin access on the cluster)? Or we are saying the mirror operation is privileged by default?

Reply:
You bring up a good point here. I would say that the user can at least construct their pull secret and put it in the expected location without needing privileged access. Our original intention with adding the pull secret to the spec was to have it directly in the config, but it quickly made the config unreadable. Eventually, we could add a flag to allow the user to specify a custom location for their pull secret. Still, we would either have to move it into an expected location or PR some of our dependencies to support this. WDYT?

Reply:
That sounds reasonable.